database/migrations/2025_04_08_194334_create_model_sync_states_table.php

```diff
@@ -15,7 +15,7 @@ public function up(): void
             $table->id();
             $table->morphs('syncable');
             $table->string('source')->nullable();
-            $table->uuid('client_generated_id')->index()->nullable();
+            $table->string('client_generated_id')->index()->nullable();
```

Comment:
```suggestion
        $table->uuid('client_generated_id')->nullable()->index();
        $table->unsignedInteger('device_id')->index();

        // Add a unique constraint across syncable_type, syncable_id, device_id, and client_generated_id
        $table->unique(['syncable_type', 'syncable_id', 'device_id', 'client_generated_id'], 'unique_syncable_device_client_id');
```

I guess this would be a good time to add support for : https://github.com/whilesmartphp/laravel-devices

The device UUID can be stored in the token field.